### PR TITLE
Language icon is now hidden unless you know those, adds linguist quirk

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request

language icons are no longer to be displayed unless you know that language.

## Why It's Good For The Game

Mateknowledge bad. Showing language icons is a kind of giving you a metaknowledge about the feature of a language.
~~also, multilingual isn't very helpful for RP since the quirk is too unreliable trait and extremely gimmicky. I don't think such randomness is good to be a part of quirk, so replace it into the new quirk.~~

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

* sayer
image

* listener
image

from chat

--------------

* sayer
image

* listener
image

from runechat

codespeak + linguist
----

* sayer
image

* listener
image

voltatic - I know it so I see the icon
draconic - I don't know, but I have linguist quirk so I can recognise
codespeak - what the hell is that?

* sayer
image

* listener
image

</details>

## Changelog
:cl:
tweak: language icons will not be displayed if you can't understand that language. only curators or linguist quirk holder will see lang icons. specific languages don't follow this rule.
add: linguist quirk. Having this quirk allows you to recognize the language icon.
/:cl:

### CODE REVIEW

**Comment by itsmeow** [comment]
I would prefer a quirk to know per language instead of recognizing icons

**Comment by EvilDragonfiend** [comment]
> I would prefer a quirk to know per language instead of recognizing icons

knowing per language feels like bloating. I will wait for more opinions from other maintainers though.
and recognising icons is a different feature.

**Comment by Archanial** [comment]
You can recognize languages from the way they sound.

**Comment by EvilDragonfiend** [comment]
> You can recognize languages from the way they sound.

that doesn't mean showing the icons is necessary.

**Comment by itsmeow** [comment]
I am completely fine with the icon removal part of this, in fact you could remove them entirely and not have the quirk and I'd be fine with merging, but as it is I don't like how multilingual was replaced (although I do agree it is a problem), there should be quirks to know Moffic, Buzzwords, Uncommon, etc separate, or a selection menu or something for languages.

**Comment by Rukofamicom** [comment]
> You can recognize languages from the way they sound.

I second this tbh. I'm usually all for restricting meta-knowledge, but I have a hard time classifying this as metaknowledge at all. I can distinctly recognize and differentiate well over a dozen languages by how they sound without knowing the language at all. I don't know Spanish I can audibly tell it apart from French and Italian which I also don't know. 

> that doesn't mean showing the icons is necessary

I'm pretty sure they meant IRL, as in the icons aren't really immersion breaking.

---

The only case to be made for immersion breaking would be coded languages based on existing ones. Syndicate codespeak, for example, should appear as ordinary galactic common, but there's a lot of work to be done on it to make it actually appear like someone isn't just spewing total nonsense.

**Comment by PowerfulBacon** [comment]
I like this

**Comment by EvilDragonfiend** [comment]
> I am completely fine with the icon removal part of this, in fact you could remove them entirely and not have the quirk and I'd be fine with merging, but as it is I don't like how multilingual was replaced (although I do agree it is a problem), there should be quirks to know Moffic, Buzzwords, Uncommon, etc separate, or a selection menu or something for languages.

We should have a language menu like what Skyrat has, but It'd be too big refactoring, so I will just put this as multi-language quirks instead. That'll make the code a bit uglier, but we can clean it later.


> I can distinctly recognize and differentiate well over a dozen languages by how they sound without knowing the language at all. 

> the icons aren't really immersion breaking

Okay, let's say a case: you have 2 characters. John, Alicide

John is a curator that he knows everything. John is your main, so you've been a player who know languages.
Alicide is your alt mothperson, an assistent, a very young girl. She doesn't know a lot about the station and its crews. but somehow she can just recognise it's draconic/voltaic/beachtongue since icon indicates it. You can pretend that you don't know it, but your immersion is already broken, I believe.

so the problem is the game just lets you know 'this is foreign language!!!!!!!' through showing the icons to you. I believe it is fine as long as you can really recognise by how it sounds through texts, not by icons.

**Comment by EvilDragonfiend** [comment]
image

This is the full list of languages. It's a bit meh to put each of them as a single quirk... I will revert the multilingual for now, but it should be removed at the end.

**Comment by Rukofamicom** [comment]
> but somehow she can just recognise their draconic/voltaic/beachtongue since icon indicates it. You can pretend that you don't know it, but your immersion is already broken (I do, and I believe it happens to other people).

> so the problem is the game just lets you get 'this is foreign language!

You aren't getting what I'm saying at all, or at least your response doesn't make me think you are. 

* Voltaic would be static and zaps at worst or perhaps just electronic sounding. The point is the sound is produced by an electric current, it would be immediately recognizable as unique by the sound of the language itself. Even to someone who had only heard it once in that moment would be able to recognize its uniqueness compared to other languages they have never heard. 
 
* Draconic could go in any number of directions, but since lizard anatomy is already at odds with common, I don't doubt it would still be very unique at first listen and completely distinct.

* Beachtongue is a **good** case for removal of the icon since it is essentially just heavy slang, and not actually supposed to be a language based on what little is present. But at the same time beachgoers are so nonsensical they border on immersion breaking as it is.

**Comment by EvilDragonfiend** [comment]
In general, I hate giving any hint of such information. If I give an example of a MMORPG wow
```
OrcKing: [Orcish] Tau rok tar gar, stobu taro.
OrcKing: [Orcish] Dama garo rok!
```
This is the most standards, but I don't like how games indicate languages with such information
```
OrcKing: Tau rok tar gar, stobu taro.
OrcKing: Dama garo rok!
```
This is my preference.

I know you're saying unearthly sounds can be insufficient to express how these sound, but I think that should be done with unique letters rather than just alphabets.

**Comment by Archanial** [comment]
> that doesn't mean showing the icons is necessary

doesn't mean it's metaknowledge (or mateknowledge)

**Comment by Rukofamicom** [comment]
> In general, I hate giving any hint of such information.

The icon replaces what pure text cannot, and that's all. It's not some magic metaknowledge, it's indicating something that would be immediately apparent to anyone actually hearing the language with their ears instead of their eyes.

**Comment by EvilDragonfiend** [comment]
(Writing in mobile)
You have verb type in your tongue and it already determines how these sound.
Indicating lang icon is not that different from showing commns channel from your chat. Hearing someone talking through syndi channel and saying they're using syndi channel is metaknowledge, right? The icon is not difterent from showing "[draconic]" in their chat and it lets you know they're talking in specific language even if you don't know that language.
So I don't think it's been doing what it is supposed to do.

**Comment by EvilDragonfiend** [comment]
To say about the syndi channel case, I am thinking a PR that all channels you hear but you don't have access to are changed into "[unknown]" rather than knowing the specific channel. This has the similar context with this too.

**Comment by Rukofamicom** [comment]
> You have verb type in your tongue and it already determines how these sound.

This shows the same verb regardless of language...

> Hearing someone talking through syndi channel and saying they're using syndi channel is metaknowledge, right?

It is metaknowledge to know something was said on a specific channel? That's completely asinine if true, and I don't think it is.

What is metaknowledge **is seeking access to the syndicate channel** before knowing it exists, not being able to recognize individual channels on comms.

**Comment by EvilDragonfiend** [comment]
> This shows the same verb regardless of language...

That'd be the main point of unearthy sounds you hear mostly.

> It is metaknowledge to know something was said on a specific channel? That's completely asinine if true, and I don't think it is.

You can say what's been talked, but knowing the specific channel name is metaknowledge. How can you know something's been talked through syndicate channel when you just 'hear" those?
I know a case someone who heard a syndi channel from someone and shouting that the specific one was talking through syndi channel, and they got an admin note.

**Comment by Rukofamicom** [comment]
> If you are a cargo tech but standing next to doctor and hearing their talking channel will be seen as [unknown] instead of [medical]
This has the similar context with this too.

.. Jesse what the fuck are you on?!

The cargo tech won't see [medical] in this case, they will fail to see the comms at all!

**Comment by EvilDragonfiend** [comment]
> > If you are a cargo tech but standing next to doctor and hearing their talking channel will be seen as [unknown] instead of [medical]
> > This has the similar context with this too.
> 
> _Jesse what the fuck are you on?_
> 
> The cargo tech won't see [medical] in this case, they will fail to see the comms at all. Go test that right now, stand next to someone and use a radio channel they don't have access to. They will see your text italicized to indicate you are speaking into a radio, but they will not see the radio message itself or the channel name you spoke into.

You can hear those if you're on the same tile, or next to them when the message is loud mode.

**Comment by EvilDragonfiend** [comment]
Anyway, the main idea for both of them is "don't let you know unless you find it yourself." And I defined it as a feature of metaknowledge stuff.
Unless you recognise this is a foreign language by yourself, don't tell it is a foreign language. Unless you have the correct access to a radio channel, don't tell which radio channel it is.

**Comment by Rukofamicom** [comment]
I'm giving up I think, but one last bit:

> Unless you recognise this is a foreign language by yourself, don't tell it is a foreign language

This is so ass-backwards. Unless you, individually as a player, can recognize the specific patterns **via metaknowledge** you can no longer tell any languages apart. You are not preventing metaknowledge, you are *forcing* it to be required in order to tell languages apart.

**Comment by EvilDragonfiend** [comment]
Also, as a RP aspect, you won't automatically evaluate someone if they're saying in foreign language or extremely broken language due to brain trauma, or pretend to say a not-language. Cultist will be capable of saying narsian in disguise.

**Comment by itsmeow** [comment]
Thinking about it more I'm fine with the linguist quirk, but I would like an alternative to multilingual, or at minimum add it back.

**Comment by EvilDragonfiend** [comment]
> Thinking about it more I'm fine with the linguist quirk, but I would like an alternative to multilingual, or at minimum add it back.

It's added back as what it was. It's really bad to add each language of what multilingual picks.

image

File: code/datums/traits/good.dm [hunk 1]
```diff
@@ -96,6 +96,14 @@
 	)
 	H.equip_in_one_of_slots(B, slots , qdel_on_fail = TRUE)
 
+/datum/quirk/linguist
+	name = "Linguist"
+	desc = "Although you don't know every language, you have spent your interest into languages. You can recognise linguistic features for each language."
```
Comment (itsmeow) [inline code/datums/traits/good.dm]: Although you don't know every language, your intense interest in languages allows you to recognize the features of most languages.

File: code/datums/traits/good.dm [hunk 2]
```diff
@@ -96,6 +96,14 @@
...
+	value = 1
+	mob_trait = TRAIT_LINGUIST
+	gain_text = "span class='notice'>You can recognise linguistic features for each language.</span>"
```
Comment (itsmeow) [inline code/datums/traits/good.dm]: You can now recognize the linguistic features of every language

**Review by itsmeow — CHANGES_REQUESTED** [review]

**Review by itsmeow — APPROVED** [review]
I am happy with this, I don't think a meta-icon telling someone what language is being spoken makes a lot of sense generally, and identifying a language by its looks is certainly a skill the player can learn if they want, given that previously you would automatically know every language's look anyway this is still an improvement if we are complaining about metagaming here, it makes it harder.

**Pull request merged**